apps/omg_child_chain/test/omg_child_chain/release_tasks/set_fee_feed_adapter_opts_test.exs

```diff
@@ -0,0 +1,119 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.ReleaseTasks.SetFeeFeedAdapterOptsTest do
+  use ExUnit.Case, async: true
+  alias OMG.ChildChain.Fees.FeedAdapter
+  alias OMG.ChildChain.ReleaseTasks.SetFeeFeedAdapterOpts
+
+  @app :omg_child_chain
+  @config_key :fee_adapter
+  @env_fee_adapter "FEE_ADAPTER"
+  @env_fee_feed_url "FEE_FEED_URL"
+  @env_fee_change_tolerance_percent "FEE_CHANGE_TOLERANCE_PERCENT"
+  @env_stored_fee_update_interval_minutes "STORED_FEE_UPDATE_INTERVAL_MINUTES"
+
+  setup do
+    original_config = Application.get_all_env(@app)
+
+    on_exit(fn ->
+      # Delete all related env vars
+      :ok = System.delete_env(@env_fee_adapter)
+      :ok = System.delete_env(@env_fee_feed_url)
+      :ok = System.delete_env(@env_fee_change_tolerance_percent)
+      :ok = System.delete_env(@env_stored_fee_update_interval_minutes)
+
+      # configuration is global state so we reset it to known values in case it got fiddled before
+      :ok = Enum.each(original_config, fn {key, value} -> Application.put_env(@app, key, value, persistent: true) end)
+    end)
+
+    :ok
+  end
+
+  test "sets the fee adapter to FeedAdapter and configure with its related env vars" do
+    :ok = System.put_env(@env_fee_adapter, "feed")
+    :ok = System.put_env(@env_fee_feed_url, "http://example.com/fee-feed-url")
+    :ok = System.put_env(@env_fee_change_tolerance_percent, "10")
+    :ok = System.put_env(@env_stored_fee_update_interval_minutes, "600")
+    config = SetFeeFeedAdapterOpts.load([], [])
+
+    {adapter, opts: adapter_opts} = config[@app][@config_key]
+    assert adapter == FeedAdapter
+    assert adapter_opts[:fee_feed_url] == "http://example.com/fee-feed-url"
+    assert adapter_opts[:fee_change_tolerance_percent] == 10
+    assert adapter_opts[:stored_fee_update_interval_minutes] == 600
+  end
+
+  test "raises an ArgumentError when FEE_CHANGE_TOLERANCE_PERCENT is not a stingified integer" do
+    :ok = System.put_env(@env_fee_adapter, "feed")
+
+    :ok = System.put_env(@env_fee_change_tolerance_percent, "1.5")
+    assert_raise ArgumentError, fn -> SetFeeFeedAdapterOpts.load([], []) end
```

Comment:
is this raising with a `message` key? do you want to check if's returning your message?

Reply:
It was a plain `ArgumentError` from `String.to_integer/1`. I've updated it to include a message + assertions though! https://github.com/omisego/elixir-omg/pull/1385/commits/f2157dccd7674d6fc800b77ccef788d221f38bca